thread: k_float_enable() API can't build on x86_64 platforms, fix that API and macro documentation

**Describe the bug**
1. In https://docs.zephyrproject.org/latest/reference/kernel/other/float.html we have description of the `k_float_enable()` API, but there is no detailed description in the documentation about that API. At the same time we don't have description about macro `K_SSE_REGS`
Both of them are highlighted red, that means no links for that objects.

2. When I try to use on qemu_x86_64 API `k_float_enable()` I receive an error ```undefined reference to `arch_float_enable'```

**To Reproduce**
1. Change hello_world sample
```
#include <zephyr.h>
#include <sys/printk.h>

void main(void)
{
	k_float_enable(_current, K_FP_REGS);
	printk("Hello World! %s\n", CONFIG_BOARD);
}
```
2. Add into prj.conf
```
CONFIG_FPU=y
CONFIG_FPU_SHARING=y
```
3. $ west build -p -b qemu_x86_64 samples/hello_world/
You will see an error:
```undefined reference to `arch_float_enable'```

**At the same time it works on qemu_x86**
$ west build -p -b qemu_x86 samples/hello_world/

**Expected behavior**
API k_float_enable() must work for x86_64.

**Impact**
I can't use macro `k_float_enable()` in an already-created x86 thread to pretag itself on x86_64 platforms (tested on qemu_x86_64, up_squared, ehl_crb).
To do this on x86_64, I have to dynamically create a thread and pretag it by passing the` K_FP_REGS` or `K_SSE_REGS` option to `k_thread_create()` (tested, it works on qemu_x86_64, up_squared, ehl_crb).
I think both variants should work on x86_64 like it is described in https://docs.zephyrproject.org/latest/reference/kernel/other/float.html (Section x86 architecture)

**Environment (please complete the following information):**
 - OS: Linux Fedora 34 (Workstation Edition)
 - Toolchain Zephyr SDK 0.13.0
 - Commit SHA 2b91ebe16e25e5ce17270ea36b88fbb34f5f28ed

--- a/arch/x86/core/intel64/thread.c
+++ b/arch/x86/core/intel64/thread.c
@@ -61,3 +61,15 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	thread->arch.flags = X86_THREAD_FLAG_ALL;
 	thread->switch_handle = thread;
 }
+
+int arch_float_disable(struct k_thread *thread)
+{
+	/* x86-64 always has FP/SSE enabled so cannot be disabled */
+	return -ENOTSUP;
+}
+
+int arch_float_enable(struct k_thread *thread, unsigned int options)
+{
+	/* x86-64 always has FP/SSE enabled so nothing to do here */
+	return 0;
+}

--- a/include/kernel.h
+++ b/include/kernel.h
@@ -214,7 +214,15 @@ extern void k_thread_foreach_unlocked(
 /* x86 Bitmask definitions for threads user options */
 
 #if defined(CONFIG_FPU_SHARING) && defined(CONFIG_X86_SSE)
-/* thread uses SSEx (and also FP) registers */
+/**
+ * @brief FP and SSE registers are managed by context switch on x86
+ *
+ * @details
+ * This option indicates that the thread uses the x86 CPU's floating point
+ * and SSE registers. This instructs the kernel to take additional steps to
+ * save and restore the contents of these registers when scheduling
+ * the thread. No effect if @kconfig{CONFIG_X86_SSE} is not enabled.
+ */
 #define K_SSE_REGS (BIT(7))
 #endif
 #endif

--- a/kernel/include/kernel_arch_interface.h
+++ b/kernel/include/kernel_arch_interface.h
@@ -181,8 +181,9 @@ void arch_switch_to_main_thread(struct k_thread *main_thread, char *stack_ptr,
  * @note For ARM architecture, disabling floating point preservation may only
  * be requested for the current thread and cannot be requested in ISRs.
  *
- * @retval 0       On success.
- * @retval -EINVAL If the floating point disabling could not be performed.
+ * @retval 0        On success.
+ * @retval -EINVAL  If the floating point disabling could not be performed.
+ * @retval -ENOTSUP If the operation is not supported
  */
 int arch_float_disable(struct k_thread *thread);
 
@@ -192,7 +193,7 @@ int arch_float_disable(struct k_thread *thread);
  * The function is used to enable the preservation of floating
  * point context information for a particular thread.
  * This API depends on each architecture implimentation. If the architecture
- * does not support enableing, this API will always be failed.
+ * does not support enabling, this API will always be failed.
  *
  * The @a options parameter indicates which floating point register sets will
  * be used by the specified thread. Currently it is used by x86 only.
@@ -200,8 +201,9 @@ int arch_float_disable(struct k_thread *thread);
  * @param thread  ID of thread.
  * @param options architecture dependent options
  *
- * @retval 0       On success.
- * @retval -EINVAL If the floating point enabling could not be performed.
+ * @retval 0        On success.
+ * @retval -EINVAL  If the floating point enabling could not be performed.
+ * @retval -ENOTSUP If the operation is not supported
  */
 int arch_float_enable(struct k_thread *thread, unsigned int options);
 #endif /* CONFIG_FPU && CONFIG_FPU_SHARING */

